app/views/books/_form.html.erb

```diff
@@ -19,7 +19,7 @@
     </fieldset>
     <fieldset>
         <%= f.label :location %>
-        <%= f.text_field :location %>
+        <%= f.text_field :location, disabled: disabled %>
```

Comment:
All the fields that librarians can't edit are disabled. The value here is a boolean variable that's being set on lines 1 & 2, based on role of user.